lambda_backend/predict_microservice/base_inference.py

```diff
@@ -0,0 +1,144 @@
+import cv2
+import numpy as np
+from configparser import RawConfigParser
+
+DEF_SCORE = 0.3
+DEF_NMS = 0.4
+
+class BaseInference:
+    """
+    Base Inference class for predictions
+    
+    ...
+
+    Attributes
+    ----------
+    weight_path : str
+        path to the .weight file
+    config_path : str
+        path to the .cfg file
+    classes : list
+        names of classes detected
+    score_thresh : float
+        threshold to classify object as detected
+    nms_threshold : float
+        threshold for non-max suppression
+
+    Methods
+    -------
+    initialize()
+        Initializes neural network using given weight and config
+    read_config()
+        Reads config for trained height and width
+    run()
+        Obtains predicted boxes 
+    """
+
+    def __init__(self, weight_path, config_path, classes, score_thresh=None, nms_thresh=None):
+        self.weight_path = weight_path
+        self.config_path = config_path
+        self.classes = classes
+        self.net = None
+        self.score_thresh = score_thresh if score_thresh is not None else DEF_SCORE
+        self.nms_thresh = nms_thresh if nms_thresh is not None else DEF_NMS
+    
+    def initialize(self):
+        # Load Yolo
+        self.net = cv2.dnn.readNet(
+            self.weight_path,
+            self.config_path
+        )
+        layer_names = self.net.getLayerNames()
+        self.output_layers = [layer_names[i[0] - 1] for i in self.net.getUnconnectedOutLayers()]
+    
+    def read_config(self):
```

Comment:
Similarly for this too. Maybe can put assert to check `net.height` and `net.width` are present in .cfg